cmds/core/sync/sync_linux.go

```diff
@@ -48,11 +48,11 @@ func doSyscall(syscallNum uintptr, args []string) error {
 	return nil
 }
 
-func sync(args []string) error {
+func sync(data, filesystem bool, args []string) error {
 	switch {
-	case *data:
+	case data:
 		return doSyscall(unix.SYS_FDATASYNC, args)
-	case *filesystem:
+	case filesystem:
 		return doSyscall(unix.SYS_SYNCFS, args)
```

Comment:
so here's my suggestion. Create a file, 
sync_linux.go
with a function in it, 
```
func syncfs(args []string) error {
return doSyscall(unix.SYS_SYNCFS, args)
}
```

create a file, 
sync_other.go, with the function
```
//go:build !linux
func syncfs(_ []string) error {
return fmt.Errorf("syncfs is not available on %v:%w", runtime.GOOS, os.ErrInvalid)
}
```

also, if a file is called sync_linux.go, it does not need a build tag for linux.

Sound OK?

Reply:
I made it a bit different, but more or less the same idea. Without magic numbers and runtime. It should work for other Unix systems right now.